src/components/payroll/PaymentMethodPicker.js

```diff
@@ -0,0 +1,68 @@
+import React, { useEffect } from 'react';
+import { connect } from 'react-redux';
+import { bindActionCreators } from 'redux';
+import { SelectInput } from '@openimis/fe-core';
+import { injectIntl } from 'react-intl';
+import { fetchPaymentMethods } from '../../actions';
+
+function PaymentMethodPicker({
+  value,
+  label,
+  onChange,
+  readOnly = false,
+  withNull = false,
+  nullLabel = null,
+  withLabel = true,
+  required = false,
+  paymentMethods,
+  fetchPaymentMethods,
+}) {
+  const fetch = () => {
+    fetchPaymentMethods({});
+  };
+
+  useEffect(() => {
+    fetch();
+  }, []);
+
+  const options = Array.isArray(paymentMethods) && paymentMethods !== undefined ? [
+    ...paymentMethods.map((paymentMethod) => ({
+      value: paymentMethod.name,
+      label: paymentMethod.name,
+    })),
+  ] : [];
+
+  if (withNull) {
+    options.unshift({
+      value: null,
+      label: nullLabel || '',
+    });
+  }
+
+  return (
+    <SelectInput
+      module="payroll"
+      label={withLabel && label}
+      options={options}
+      value={value}
+      onChange={onChange}
+      readOnly={readOnly}
+      required={required}
+    />
+  );
+}
+
+const mapStateToProps = (state) => ({
+  rights: !!state.core && !!state.core.user && !!state.core.user.i_user ? state.core.user.i_user.rights : [],
+  confirmed: state.core.confirmed,
+  fetchingPaymentMethods: state.payroll.fetchingPaymentMethods,
+  errorPaymentMethods: state.payroll.errorPaymentMethods,
+  fetchedPaymentMethods: state.payroll.fetchedPaymentMethods,
+  paymentMethods: state.payroll.paymentMethods,
+});
+
+const mapDispatchToProps = (dispatch) => bindActionCreators({
+  fetchPaymentMethods,
+}, dispatch);
+
+export default injectIntl(connect(mapStateToProps, mapDispatchToProps)(PaymentMethodPicker));
```

Comment:
Is it necessary to use _injectIntl_ here? You're providing label as a prop and it is already translated. So I think that _injectIntl_ could be removed from this Picker.

Reply:
It should not affect the component, since intl is not used "inside".

Reply:
great, I've just applied your suggestion